lib/knapsack_pro/config/env.rb

```diff
@@ -23,9 +23,10 @@ def ci_node_index
         end
 
         def ci_node_build_id
-          ENV['KNAPSACK_PRO_CI_NODE_BUILD_ID'] ||
+          env_name = 'KNAPSACK_PRO_CI_NODE_BUILD_ID'
+          ENV[env_name] ||
             ci_env_for(:node_build_id) ||
-            'missing-build-id'
+            raise("Missing environment variable #{env_name}. Read more at #{KnapsackPro::Urls::KNAPSACK_PRO_CI_NODE_BUILD_ID}")
```

Comment:
You could update [CHANGLELOG](https://github.com/KnapsackPro/knapsack_pro-ruby/blob/master/CHANGELOG.md) as part of this PR. Please also make sure you release a major version 4.0.0 because we have a breaking change here. If someone uses CI provider without out-of-the-box CI build ID support then we could break their CI builds with this change when they upgrade the gem.